File: modul_praktikum_python/contoh_code/python/diskriminan_kuadrat.py
"""Menentukan jenis akar persamaan kuadrat dari diskriminan."""


def main() -> None:
    print("Menentukan Jenis Akar Persamaan Kuadrat")
    print("      Persamaan Umum : ax^2+bx+c       ")
    print()
    a = float(input("a = "))
    b = float(input("b = "))
    c = float(input("c = "))
    print()
    d = (b * b) - (4 * a * c)
    print("Nilai Diskriminan :", d)
    print("Jenis Akar : ", end="")
    if d == 0:
        print("Kembar", end="")
    elif d > 0:
        print("Berlainan", end="")
    elif d < 0:
        print("imajiner berlainan", end="")
    print()
    input("Tekan Enter untuk melanjutkan...")


if __name__ == "__main__":
    main()

File: modul_praktikum_python/contoh_code/pascal/diskriminan_kuadrat.pas
uses crt; 
var 
  a,b,c,D : real; 
begin 
  writeln(‘Menentukan Jenis Akar Persamaan Kuadrat’); 
  writeln(‘      Persamaan Umum : ax^2+bx+c       ’); 
  writeln; 
  write(‘a = ’);readln(a); 
  write(‘b = ’);readln(b); 
  write(‘c = ’);readln(c); 
  writeln; 
  D:=(b*b)-(4*a*c); 
  writeln(‘Nilai Diskriminan : ’,D);  
  write(‘Jenis Akar : ’); 
  if D=0 then  
     write(‘Kembar’) 
  else 
    if D>0 then  
     write(‘Berlainan’) 
    else 
    if D<0 then  
     write(‘imajiner berlainan’); 
  readkey; 
end.
